src/sidebar/components/sidebar-content.js

```diff
@@ -252,8 +250,14 @@ function SidebarContentController(
   // Re-fetch annotations when focused group, logged-in user or connected frames
   // change.
   $scope.$watch(
-    () => [groups.focused().id, store.profile().userid, ...store.searchUris()],
-    ([currentGroupId], [prevGroupId]) => {
+    () => [groups.focused(), store.profile().userid, ...store.searchUris()],
+    ([currentGroup], [prevGroup]) => {
+      // If the currentGroup is null, clear the selected annotations and return immediatly.
+      if (!currentGroup) {
```

Comment:
This is necessary so that we don't load the annotations while the groups are still loading and also don't get deeper into this function when the focused group is null. By clearing the focused group we are able to prevent any load that may happen before the groups have been re-grabbed by the groups service. With the changed you proposed below, I run into the race condition described below this logic. The userid changes and the annotations are reloaded for the open group, then when the private group is focused on, the private group annotations are skipped because an annotation load is already in progress. I think clearing out the groups is a fairly simple way to handle this but I see your point about it being a bit fragile. I'll have to think about this a little more. Perhaps we can sync up and brainstorm some ideas tomorrow? I think there are a couple options here:
1) Re-think 'watching' these values and place this logic in the functions that trigger those changes. Perhaps moving the logic surrounding the loading annotations into a service or util.
2) Storing the state of when the group service has completed reloading the groups and only reload annotations when it's finished.
3) Add a more official clear state method into the groups service and call that instead of clearing the groups on the store directly.

To clarify the issue we are trying to resolve here, I think it runs deeper than just direct-linked groups, it really is any group that will change from being fetchable to not being fetchable (or vise versa) when the user logs in or out is broken due to this race condition.